### PR TITLE
Adopt dynamic calculated fees

### PR DESCRIPTION
**Pull request by jrwbabylonchain**
Pending merge of this PR https://github.com/babylonchain/btc-staking-ts/pull/24 then we can bump the ts lib version here

But the implementation needed in this repo is pretty much ready for review

### CODE REVIEW

**Review by jeremy-babylonchain — APPROVED** [review]

**Comment by jeremy-babylonchain** [comment]
looks good to me. will help test again when ts lib is ready

File: src/app/components/Delegations/Delegations.tsx [hunk 1]
```diff
@@ -218,57 +212,59 @@ export const Delegations: React.FC<DelegationsProps> = ({
...
         delegation.stakingTx.outputIndex,
       );
     }
 
     // Sign the withdrawal transaction
     let withdrawalTransaction: Transaction;
     try {
-      withdrawalTransaction = await signPsbtTx(unsignedWithdrawalTx.toHex());
+      const { psbt } = withdrawPsbtTxResult;
+      withdrawalTransaction = await signPsbtTx(psbt.toHex());
```
Comment (vitsalis) [inline src/app/components/Delegations/Delegations.tsx:262]:
```suggestion
      withdrawalTransaction = await signPsbtTx(withdrawPsbtTxResult.psbt.toHex());
```

File: src/utils/signForm.ts [hunk 2]
```diff
@@ -53,38 +49,40 @@ export const signForm = async (
...
-      unbondingScript,
-      slashingScript,
+    const netWorkFee = await btcWallet.getNetworkFees();
+    feeRate = netWorkFee.fastestFee;
+  } catch (error) {
+    throw new Error("Cannot get network fees");
+  }
+  let unsignedStakingPsbt;
+  try {
+    const { psbt } = stakingTransaction(
```
Comment (vitsalis) [inline src/utils/signForm.ts:61]:
```suggestion
    unsignedStakingPsbt = stakingTransaction(
```

**Review by vitsalis — APPROVED** [review]
lgtm

File: src/utils/signForm.ts (hunk 2 above)
Comment (gbarkhatov) [reply src/utils/signForm.ts:61]: @vitsalis it returns
```ts
PsbtTransactionResult {
    psbt: Psbt;
    fee: number;
}
```

**Review by gbarkhatov — COMMENTED** [review]
When we click stake we produce a lot of mempool calls https://youtu.be/Vj-dAtCcl0M
1. Is this what we want?
In any case we need to
1. Disable the "stake" button, until we get the response from the wallet UI
2. Show a loading indicator instead of the text

**Comment by vitsalis** [comment]
> When we click stake we produce a lot of mempool calls https://youtu.be/Vj-dAtCcl0M
> 
>     1. Is this what we want?
>        In any case we need to

Indeed, that's not nice. It happens because we get * all * UTXOs and for each one we query to get additional information about them. A neat solution to this would be to first get the basic information about all UTXOs and then query the additional information only for those that we selected.

**Comment by jrwbabylonchain** [comment]
> When we click stake we produce a lot of mempool calls https://youtu.be/Vj-dAtCcl0M
> 
> 1. Is this what we want?
>    In any case we need to
> 2. Disable the "stake" button, until we get the response from the wallet UI
> 3. Show a loading indicator instead of the text

resolved in the latest commit by using the `validate-address` endpoint

**Pull request merged**